packages/aws-rfdk/lib/deadline/lib/worker-configuration.ts

```diff
@@ -29,6 +29,39 @@ import {
   Version,
 } from './version';
 
+/**
+ * Provider for adding user data scripts
+ * Methods of this class will be invoked in WorkerInstanceConfiguration
+ * on different stages of worker configuration
+ */
+export interface IInstanceUserDataProvider {
+  // Method that is invoked before configuration Cloud Watch Agent.
+  preCloudWatchAgent(host: IHost): void;
+  // Method that is invoked before RenderQueue configuration.
+  preRenderQueueConfiguration(host: IHost): void;
+  // Method that is invoked before Deadline worker configuration.
+  preDeadlineConfiguration(host: IHost): void;
+  // Method that is invoked after all configuration is done and worker started.
+  postWorkerLaunch(host: IHost): void;
+}
+
+export class InstanceUserDataProvider extends Construct  implements IInstanceUserDataProvider{
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+  }
+
+  preCloudWatchAgent(_host: IHost): void {
```

Comment:
We'll also want docstrings on each of these, though it can probably just be an `@inheritdoc` docstring.